lib/core/src/Test/QuickCheck/Instances/UTCTime.hs

```diff
@@ -0,0 +1,67 @@
+{-# LANGUAGE NumericUnderscores #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+-- Provides a `Arbitrary` instance for `UTCTime` that spans a wide range of
+-- dates and a mixture of time precisions.
+
+module Test.QuickCheck.Instances.UTCTime ()
+    where
+
+import Prelude
+
+import Data.Time
+    ( UTCTime (..) )
+import Data.Time.Calendar
+    ( Day (ModifiedJulianDay), toModifiedJulianDay )
+import Data.Time.Clock
+    ( diffTimeToPicoseconds, picosecondsToDiffTime, secondsToDiffTime )
+import Test.QuickCheck
+    ( Arbitrary, arbitrary, choose, oneof, shrink )
+
+-- | Approximate number of days in one thousand years.
+daysInOneThousandYears :: Integral a => a
+daysInOneThousandYears = 365 * 1000
+
+-- | The maximum number of picoseconds in one day, allowing for leap seconds.
+picosecondsInOneDay :: Integral a => a
+picosecondsInOneDay = secondsInOneDay * picosecondsInOneSecond
+
+-- | The exact number of picoseconds in one second.
+picosecondsInOneSecond :: Integral a => a
+picosecondsInOneSecond = 1_000_000_000_000
+
+-- | The maxmimum number of seconds in one day, allowing for leap seconds.
+secondsInOneDay :: Integral a => a
+secondsInOneDay = (60 * 60 * 24) + 1
+
+-- | An `Arbitrary` instance for `UTCTime` that spans a wide range of dates
+--   and a mixture of time precisions.
+instance Arbitrary UTCTime where
```

Comment:
Couldn't a newtype (e.g. `UniformUTCTime ...`) help both:

- Conveying the meaning behind this instance (more uniformly distributed than the default arbitrary)
- Avoiding the orphan instance which makes imports quite unclear

Perhaps even sticking with `genUniformUTCTime :: Gen UTCTime` since the shrinker isn't really of any use for this one. And, import this helper wherever we need to define a uniform UTCTime.

Reply:
I agree. I'll update the PR so that we define a `newtype` wrapper (as suggested), and make an instance for the `newtype` instead of for `UTCTime` directly.